### PR TITLE
Feature/view refresh vault token

### PR DESCRIPTION
**Pull request by melanahammel**
Added View Token button to drop down menu on header
Created modal that displays client token and renews the client token

### CODE REVIEW

File: app/components/Header/Header.js [hunk 1]
```diff
@@ -6,6 +6,10 @@ import * as authActions from '../../actions/authenticationActions'
 import './Header.scss'
 import '../../assets/images/cerberus-logo-narrow-off-white.svg'
 
+import * as modalActions from '../../actions/modalActions'
+// import CreateSDBoxForm from '../CreateSDBoxForm/CreateSDBoxForm'
```
Comment (sdford) [inline app/components/Header/Header.js]: I assume this line isn't needed?

File: app/components/ViewTokenModal/ViewTokenModal.js [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+
+    this.handleRenewTokenClicked = function() {
+      this.props.dispatch(authActions.refreshAuth(this.props.clientToken, '', false))
+    }.bind(this)
+  }
+
+  render() {
+    const {clientToken, dispatch, isAuthenticating} = this.props
+    let tokenExpiresDate = sessionStorage.getItem('tokenExpiresDate');
+    let entry = clientToken
```
Comment (sdford) [inline app/components/ViewTokenModal/ViewTokenModal.js]: Is this variable needed? It doesn't seem like it's used anywhere

File: app/components/ViewTokenModal/ViewTokenModal.js [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+import { connect } from 'react-redux'
+import { reduxForm, touch } from 'redux-form'
+import * as modalActions from '../../actions/modalActions'
+import * as appActions from '../../actions/appActions'
+import * as authActions from '../../actions/authenticationActions'
+import CopyToClipboard from 'react-copy-to-clipboard';
+
+import './ViewTokenModal.scss'
+
+// Some code from https://github.com/mathieudutour/react-progress-button
```
Comment (sdford) [inline app/components/ViewTokenModal/ViewTokenModal.js]: It seems like this code is removed, so this comment may not be necessary?

File: package.json [hunk 4]
```diff
@@ -28,7 +28,7 @@
     "react-addons-css-transition-group": "15.1.0",
     "react-addons-shallow-compare": "15.1.0",
     "react-copy-to-clipboard": "4.1.0",
-    "react-dom": "15.1.0",
+    "react-dom": "^15.1.0",
```
Comment (sdford) [inline package.json]: I think we want to leave the version hardcoded here (e.g. `"react-dom": "15.1.0",`)

File: app/actions/authenticationActions.js [hunk 5]
```diff
@@ -197,10 +198,13 @@ export function refreshAuth(token, redirectPath='/', redirect=true) {
         .then(function (response) {
             dispatch(handleRemoveAuthTokenTimeout())
             dispatch(handleRemoveSessionWarningTimeout())
-            handleUserLogin(response, dispatch, false)
-            if (redirect) {
-                hashHistory.push(redirectPath)
-            }
+            setTimeout(function(){
```
Comment (sdford) [inline app/actions/authenticationActions.js]: @fieldju Is this just to simulate a loading effect in the view?

Comment (fieldju) [inline app/actions/authenticationActions.js]: @sdford yes

File: app/actions/authenticationActions.js [hunk 6]
```diff
@@ -197,10 +198,13 @@ export function refreshAuth(token, redirectPath='/', redirect=true) {
...
+                handleUserLogin(response, dispatch, false)
+                if (redirect) {
+                    hashHistory.push(redirectPath)
+                }
+            }, 3000);
```
Comment (fieldju) [inline app/actions/authenticationActions.js]: This was a little too long in the demo, I think.

File: app/components/ViewTokenModal/ViewTokenModal.scss [hunk 7]
```diff
@@ -280,3 +280,10 @@
     transform-origin: center center;
   }
 }
+
+.view-token-modal {
```
Comment (fieldju) [inline app/components/ViewTokenModal/ViewTokenModal.scss]: I don't think this is right since it didn't work in the demo

File: startLocalEnv.sh [hunk 8]
```diff
@@ -21,9 +21,9 @@ echo ""
 echo ""
 echo "Starting vault locally"
 
-vault server -config=local-persist.hcl 1>logs/vault.stdout.log 2>&1 &
+#vault server -config=local-persist.hcl 1>logs/vault.stdout.log 2>&1 &
```
Comment (fieldju) [inline startLocalEnv.sh]: need to revert the changes in this file

Comment (melanahammel) [inline startLocalEnv.sh]: So we want lines 24 and 26 commented out?
